packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,208 @@
+  final ValueChanged<Key> onValueChanged;
+
+  List<Widget> _buildSegmentedControlButtons() {
+    List<Widget> buttons = new List<Widget>();
+
+    for (int i = 0; i < children.length; i += 1) {
+      buttons.add(new Expanded(
+          child: new _SegmentedControlButton(
+              children.elementAt(i),
+              children.elementAt(i).key,
+              onValueChanged,
+              value,
+              false,
+              i == 0,
+              i == children.length - 1)));
+
+      if (i != children.length - 1) {
+        buttons.add(new _SegmentedControlButton(
+            new Container(
+                decoration:
+                    const BoxDecoration(color: CupertinoColors.activeBlue),
+                constraints:
+                    const BoxConstraints(maxWidth: 2.0, minWidth: 2.0)),
+            children.elementAt(i).key,
+            onValueChanged,
+            value,
+            true,
+            false,
+            false));
+      }
+    }
+    return buttons;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      margin: _kHorizontalItemPadding,
+      child: new Align(
+        child: new Container(
+          decoration: const ShapeDecoration(
+            color: CupertinoColors.white,
+            shape: const RoundedRectangleBorder(
+              side: BorderSide(
+                  color: CupertinoColors.activeBlue,
+                  width: 2.0,
+                  style: BorderStyle.solid),
+              borderRadius: BorderRadius.all(_kDefaultCornerRadius),
+            ),
+          ),
+          child: new Row(
+              mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+              children: _buildSegmentedControlButtons()),
+        ),
+      ),
+    );
+  }
+}
+
+class _SegmentedControlButton extends StatefulWidget {
+  _SegmentedControlButton(this.selectState, this.widgetKey, this.onValueChanged,
+      this.sharedState, this.isDivider, this.isFirst, this.isLast);
+
+  final Widget selectState;
+  final Key widgetKey;
+  final ValueChanged<Key> onValueChanged;
+  Key sharedState;
+  final bool isDivider;
+  final bool isFirst;
+  final bool isLast;
+
+  @override
+  _SegmentedControlButtonState createState() => _SegmentedControlButtonState();
+}
+
+class _SegmentedControlButtonState extends State<_SegmentedControlButton> {
+  static const Radius defaultInnerCornerRadius = Radius.circular(1.0);
+  static const Color onPressColor = const Color(0x33007aff);
+  bool isPressed = false;
+
+  _SegmentedControlButtonState();
+
+  static const BorderRadius leftmostWidgetBorder = const BorderRadius.only(
+      topLeft: defaultInnerCornerRadius, bottomLeft: defaultInnerCornerRadius);
+  static const BorderRadius rightmostWidgetBorder = const BorderRadius.only(
+      topRight: defaultInnerCornerRadius,
+      bottomRight: defaultInnerCornerRadius);
+
+  Color _calculateBackgroundColor(bool isSelected) {
```

Comment:
This could be a getter `_backgroundColor` if you move the isSelected comparison inside instead of passing it in.